src/icons/file/IconFileAudio.js

```diff
@@ -1,34 +1,49 @@
 // @flow
 import * as React from 'react';
+import uniqueId from 'lodash/uniqueId';
 
 import AccessibleSVG from '../accessible-svg';
+import OutlineFilter from '../OutlineFilter';
+import type { Icon, OutlinedStyle } from '../flowTypes';
 
-type Props = {
-    className?: string,
-    height?: number,
-    /** A text-only string describing the icon if it's not purely decorative for accessibility */
-    title?: string | React.Element<any>,
-    width?: number,
+type Props = Icon & {
+    styleProps?: OutlinedStyle,
 };
 
-const IconFileAudio = ({ className = '', height = 32, title, width = 32 }: Props) => (
-    <AccessibleSVG
-        className={`icon-file-audio ${className}`}
-        height={height}
-        title={title}
-        viewBox="0 0 32 32"
-        width={width}
-    >
-        <path d="M25 27H7V5h13l5 5v17z" fill="#fff" />
-        <path
-            d="M20 4H7a.94.94 0 0 0-1 1v22a.94.94 0 0 0 1 1h18a.94.94 0 0 0 1-1V9.9zm5 23H7V5h13v4a.94.94 0 0 0 1 1h4z"
-            fill="#955ca5"
-        />
-        <path
-            d="M19 20.1V17h-6v4.5a1.5 1.5 0 1 1-1.5-1.5.9.9 0 0 1 .5.1V13h8v8.5a1.5 1.5 0 1 1-1.5-1.5.9.9 0 0 1 .5.1zM13 16h6v-2h-6z"
-            fill="#955ca5"
-        />
-    </AccessibleSVG>
-);
+class IconFileAudio extends React.Component<Props> {
+    componentWillMount() {
+        this.outlineId = uniqueId('outline-');
```

Comment:
Is there any way to create a HOC that does all this common mounting logic or a utility function that creates the filterProp below to reduce repetition?

Reply:
How about a hook? Could do that as a followup when we upgrade. Seems like a great use case.

Reply:
https://github.com/box/box-ui-elements/blob/master/src/icons/accessible-svg/AccessibleSVG.js#L14

Can we re-use?

Reply:
Like this:

```js
function useStableInstanceId(prefix) {
  return React.useCallback(() =>
    lodash.uniqueId(prefix),
    [prefix]
  )
}
```

```jsx
function MyIcon() {
  const [id] = useStableInstanceId('outline-')
  return <svg id={id}>{/* ... */}</svg>
}

function MyOtherIcon() {
  const [id] = useStableInstanceId('box-')
  return <svg id={id}>{/* ... */}</svg>
}
```

Reply:
AccessibleSVG already has an id, and we can add an outlineId, and just pass those down to the svg without increasing complexity

Reply:
Is the 1st child for AccessibleSVG always going to be `<OutlineFilter id={this.outlineId} styleProps={styleProps} />` for certain stuff? Can just use a prop to determine and have that code inside AccessibleSVG itself unless we prefer another level of indirection.

<AccessibleSVG>
<OutlinedSVG>
...
</OutlinedSVG>
</AccessibleSVG>

Reply:
I set up a `IconFileBase` component to remove some redundancy in all the file icons. Then I created an `OutlinedSVG` component which removed even more. Take a look at the PR now ignoring whitespace `?w=1`. Should be much easier to go through now.

We could also switch to hooks at some point, although now I think there's probably less value from that kind of switch...